File: 0128-longest-consecutive-sequence/solution.py
class Solution(object):
    def longestConsecutive(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        
        setNums = set(nums)
        longestSequence = 0
        
        for num in nums:
            if num-1 not in setNums:
                cur = num
                length = 1
                while cur + 1 in setNums:
                    length += 1
                    cur += 1

                longestSequence = max(longestSequence, length)
        return longestSequence

File: 0128-longest-consecutive-sequence/solution.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> seen(nums.begin(), nums.end());
        int maxLength = 0;


        for (auto num: nums) {
            int length = 0;
            if (seen.find(num - 1) == seen.end()) {
                while (seen.find(num) != seen.end()) {
                    num++;
                    length++;
                }
                if (maxLength < length) maxLength = length;
            }
        }

        return maxLength;
    }
};
